.github/scripts/compile.sh

```diff
@@ -122,7 +122,7 @@ do_build () {
     -DCMAKE_BUILD_TYPE=RelWithDebInfo \
     -DRUNTIME_CHECK="${runtime_check}" \
     -DSANITIZE_PYTHON=OFF \
-    -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
+    -DENABLE_LINKTIME_OPTIMIZATION=ON \
```

Comment:
I think we should align on the cmake name for this, CMAKE_INTERPROCEDURAL_OPTIMIZATION (so what you had originally).

See also the note about policy CMP0069 here:

https://cmake.org/cmake/help/latest/prop_tgt/INTERPROCEDURAL_OPTIMIZATION.html

Reply:
Proton uses `ENABLE_LINKTIME_OPTIMIZATION` so I used the same in skupper-router.

If you look at the Examples section of policy CMP0069 - https://cmake.org/cmake/help/latest/policy/CMP0069.html#policy:CMP0069 - it suggests the use of CMake's `check_ipo_supported()` which we use here. If `check_ipo_supported()` is true, we call `set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)`

Reply:
Last I spoke to Andrew, he was planning to switch Proton to the standard CMake name.

Reply:
Switched to use the standard CMake name `CMAKE_INTERPROCEDURAL_OPTIMIZATION`